### PR TITLE
fix(airflow): clear postgresql parent registry

### PR DESCRIPTION
**Pull request by omercnet**
## Summary
- Clear the airflow parent `postgresql.image.registry` override so it does not prepend `ghcr.io/` in front of the generated FQDN PostgreSQL repository.
- Keep `postgresql.image.repository` as the upstream `bitnamilegacy/postgresql` source path to avoid double `verity-org/` patching.

## Root Cause
After [#408](https://github.com/verity-org/verity/pull/408), main chart-integration removed the double `verity-org/` prefix but still failed airflow with `ghcr.io/ghcr.io/verity-org/bitnamilegacy/postgresql:16.1.0-debian-11-r15`. The parent chart registry scalar was composing in front of chartgen's generated FQDN repository override.

## Tests
- `go test ./internal/chartgen/...`
- `go test -tags integration ./test/chart-integration/... -run "TestLoadAllowlist|TestIsAccepted|TestProductionSKIPSYAMLIsValid|TestClassify|TestHarness"`

---
## Summary by cubic
Fixes the Airflow PostgreSQL image by clearing the parent registry, avoiding a double `ghcr.io` prefix and using the correct FQDN repo. Adds a regression test to lock the expected `airflow.postgresql` values.

- Bug Fixes
  - Set `airflow.postgresql.image.registry` to `""` so the parent registry isn’t prepended.
  - Keep `airflow.postgresql.image.repository` as `bitnamilegacy/postgresql` to use the upstream path.
  - Add a test to assert the registry is empty and the repository is `bitnamilegacy/postgresql`.

<sup>Written for commit 56069741093f082e06a4400f749f9dacec70691d. Summary will update on new commits. <a href="https://cubic.dev/pr/verity-org/verity/pull/410?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: verity.yaml [hunk 1]
```diff
@@ -323,12 +323,17 @@ chartValues:
   # gate stays clean.
   #
   # Supporting knobs:
-  # - `postgresql.image.registry: ghcr.io` keeps the parent-level
-  #   postgresql image reference composable while image discovery still
-  #   sees the upstream repository path below. Do NOT pre-prefix the
-  #   repository with `verity-org/`: chart-gen derives the patched
-  #   target from discovered images, so pre-prefixing makes the source
-  #   name `verity-org/bitnamilegacy/postgresql` and renders the broken
+  # - `postgresql.image.registry: ""` prevents the parent-level
```
[the remark this answers is not in the document]
Comment (omercnet) [reply verity.yaml:327]: Fixed in the latest push by removing the stale `ghcr.io` wording and documenting the empty-string registry approach.

File: verity.yaml [hunk 2]
```diff
@@ -345,7 +350,7 @@ chartValues:
   #   StatsD metrics are not required for the smoke test (which only
   #   verifies install + pod readiness, not metric scraping).
   airflow:
-    postgresql.image.registry: ghcr.io
+    postgresql.image.registry: ""
```
[the remark this answers is not in the document]
Comment (omercnet) [reply verity.yaml:355]: The Airflow PR smoke passed with `postgresql.image.registry: ""`, so the Bitnami helper treats it as unset without a leading slash here. Keeping this value and resolving.

File: verity.yaml [hunk 3]
```diff
@@ -323,12 +323,17 @@ chartValues:
...
-  #   postgresql image reference composable while image discovery still
-  #   sees the upstream repository path below. Do NOT pre-prefix the
-  #   repository with `verity-org/`: chart-gen derives the patched
-  #   target from discovered images, so pre-prefixing makes the source
-  #   name `verity-org/bitnamilegacy/postgresql` and renders the broken
+  # - `postgresql.image.registry: ""` prevents the parent-level
+  #   registry scalar from being prepended to the generated FQDN
+  #   repository override. Do NOT set it to `ghcr.io`: the generated
+  #   wrapper already writes an FQDN repository for this parent path,
+  #   and a non-empty registry renders the broken
```
[the remark this answers is not in the document]
Comment (omercnet) [reply verity.yaml]: Fixed in the latest push by calling out that chartgen overwrites `airflow.postgresql.image.repository` during wrapper generation.
